src/ILLink.RoslynAnalyzer/RequiresISymbolExtensions.cs

```diff
@@ -51,6 +51,11 @@ public static bool IsOverrideInRequiresScope (this ISymbol member, string requir
 
 		private static bool IsInRequiresScope (this ISymbol member, string requiresAttribute, bool checkAssociatedSymbol)
 		{
+			// Requires attribute on a type does not silence warnings that originate
+			// from the type directly.
+			if (member is ITypeSymbol)
+				return false;
+
 			if (member is ISymbol containingSymbol) {
 				if (containingSymbol.HasAttribute (requiresAttribute)
 					|| (containingSymbol is not ITypeSymbol &&
```

Comment:
Considering that containingSymbol is actually member and we already check that member is not ITypeSymbol this check can get deleted too

Reply:
Good point, fixed.

Reply:
Good point, fixed.

Reply:
Is there any enforcement that it is a member symbol? It looks like the method and callers assume it is right now, but I wonder if we should enforce that somehow.